drivers/network/netio/netio.c

```diff
@@ -0,0 +1,896 @@
+/*
+ * PROJECT:     NetIO driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     A more convenient networking (TCP/IP, UDP/IP) kernel API
+ *              introduced with NT6
+ * COPYRIGHT:   Copyright 2023-2024 Johannes Thoma <johannes@johannesthoma.com>
+ */
+
+/* We are a NT6+ driver .. */
+#undef _WIN32_WINNT
+#define _WIN32_WINNT 0x600
+#undef WINVER
+#define WINVER 0x600
+
+/* TODOs:
+	Done: Clean up socket in WskCloseSocket (have RefCount and
+	SocketGet / SocketPut functions).
+	Done: Remove unnecessary code (like Hook of completion)
+	Make it compile with MS VC as well
+	Rebase onto latest master
+	Squash commits (git rebase -i) one for tdihelpers one for netio
+
+	The whole Listen / Accept mechanism is still missing
+	Some minor functions (not used by WinDRBD) are missing
+	(like WskControlClient, WskSocketConnect, ...)
+	Raw sockets are not supported for now.
+	We should have regression tests for that...
+*/
+
+#include <ntdef.h>
+#include <wdm.h>
+#include <wsk.h>
+#include <ndis.h>
+#include <ntifs.h>
+#include <windef.h>
+/* From afd: this should be removed: */
+#include <debug.h>
+
+#include <tdi.h>
+#include <tcpioctl.h>
+#include <tdikrnl.h>
+#include <tdiinfo.h>
+#include <tdi_proto.h>
+#include <tdiconn.h>
+
+/* AFD also defines this. It is used by the tdi helpers */
+DWORD DebugTraceLevel = MIN_TRACE;
+
+#ifndef INVALID_HANDLE_VALUE
+#define INVALID_HANDLE_VALUE ((HANDLE) -1)
+#endif
+
+#ifndef TAG_AFD_TDI_CONNECTION_INFORMATION
+#define TAG_AFD_TDI_CONNECTION_INFORMATION 'cTfA'
+#endif
+
+/* AFD Share Flags */
+#define AFD_SHARE_UNIQUE		0x0L
+#define AFD_SHARE_REUSE			0x1L
+#define AFD_SHARE_WILDCARD		0x2L
+#define AFD_SHARE_EXCLUSIVE		0x3L
```

Comment:
Avoid tab here as well.
```suggestion
/* AFD Share Flags */
#define AFD_SHARE_UNIQUE     0x0L
#define AFD_SHARE_REUSE      0x1L
#define AFD_SHARE_WILDCARD   0x2L
#define AFD_SHARE_EXCLUSIVE  0x3L
```